### ISSUE TITLE
Some modules aren't compiling to the correct Lua version

### ISSUE THREAD

**Issue by LiamFry** [luarocks/luarocks#1209]
My understanding is that LuaRocks 3.3.1 no longer supports `--versioned-rocks-dir`  Furthermore, my understanding is that LuaRocks now detects the active Lua version and bases its operations on that, i.e. if Lua 5.2 is active, then rocks are installed in `/usr/local/{share,lib}/lua/5.2`

I've been using LuaRocks 2.4.4 for quite a while. A few days ago, I upgraded to LuaRocks 3.3.1  Upon updating my rocks, many are now broken. It appears that, regardless of the Lua version that is active, LuaRocks is building some - not all - of my rocks against Lua 5.3

Consider this: `lua-zip`, `luasec`, `lrexlib-pcre2` are but a sampling of the rocks which are compiling to Lua 5.3 *regardless* of the active Lau version. This renders these rocks useless in Lua 5.1 and 5.2  I know that these rocks are always compiling to 5.3 for two reasons; let's use `luasec` as an example:

First, when I attempt to use it, I get errors, as follows:

 1. Lua 5.1
- `dlopen(/usr/local/lib/lua/5.1/ssl.so, 2): Symbol not found: _luaL_checkversion_`
 2. Lua 5.2
- `dlopen(/usr/local/lib/lua/5.2/ssl.so, 6): Symbol not found: _lua_rotate`
 3. Lua 5.3
- success

... and ...

`ls -las /usr/local/lib/lua/5.*/ssl.so` -->
`-rwxr-xr-x  1 root  wheel  94736 Jul 23 03:11 /usr/local/lib/lua/5.1/ssl.so`
`-rwxr-xr-x  1 root  wheel  94736 Jul 23 03:11 /usr/local/lib/lua/5.2/ssl.so`
`-rwxr-xr-x  1 root  wheel  94736 Jul 23 03:11 /usr/local/lib/lua/5.3/ssl.so`
And each library binary-compares as identical.
This pattern - both Lua load errors and identical files - holds true for `zip.so` and `rex_pcre2.so` (and a few others).

I want to think that I am doing something wrong and that LuaRocks is not broken.
Without LuaRocks 2.4.4's `--versioned-rocks-dir` and three separate `luarocks` executables, what is the proper way to install the same rock for different versions of Lua?

**Comment by LiamFry** [comment]
The new flags `--lua-dir` and `--lua-version` introduced with LuaSocket 3.0.0 appear to have no effect, positive or negative. Again, using `luasec` as an example, with these flags set, the libraries are still compiled against Lua 5.3 and finish up being binary identical.

**Comment by daurnimator** [comment]
> The new flags `--lua-dir` and `--lua-version` introduced with LuaSocket 3.0.0 appear to have no effect, positive or negative.

Could you show an invocation that demonstrates this? I use `--lua-version` myself all the time.

**Comment by LiamFry** [comment]
I finally got things working with the `--lua-dir` and `--lua-version` flags. I honestly cannot find where those two are documented; this crumb trail _luarocks.org_ --> _Docs_ --> _Command-line Tools_ --> _luarocks_ --> _install_ is the most logical place one could find the needed docs but they are not there. I poked around other places; no success. Through a bunch of trial and error, I found the `--lua-version` also works with the `remove` command.
